cgi.go

```diff
@@ -161,23 +132,101 @@ func computeKnownVariables(request *http.Request, p *runtime.Pinner) (cArr [27]C
 		}
 	}
 
-	setKnownServerVariable(p, &cArr, serverName, reqHost)
+	registerTrustedVar(knownServerKeys["SERVER_NAME\x00"], reqHost, track_vars_array, thread)
 	if reqPort != "" {
-		setKnownServerVariable(p, &cArr, serverPort, reqPort)
+		registerTrustedVar(knownServerKeys["SERVER_PORT\x00"], reqPort, track_vars_array, thread)
+	} else {
+		registerTrustedVar(knownServerKeys["SERVER_PORT\x00"], "", track_vars_array, thread)
 	}
 
 	// Variables defined in CGI 1.1 spec
 	// Some variables are unused but cleared explicitly to prevent
 	// the parent environment from interfering.
 
-	// These values can not be override
-	setKnownServerVariable(p, &cArr, contentLength, request.Header.Get("Content-Length"))
-	setKnownServerVariable(p, &cArr, gatewayInterface, "CGI/1.1")
-	setKnownServerVariable(p, &cArr, serverProtocol, request.Proto)
-	setKnownServerVariable(p, &cArr, serverSoftware, "FrankenPHP")
-	setKnownServerVariable(p, &cArr, httpHost, request.Host) // added here, since not always part of headers
+	// These values can not be overridden
+	registerTrustedVar(knownServerKeys["CONTENT_LENGTH\x00"], request.Header.Get("Content-Length"), track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["GATEWAY_INTERFACE\x00"], "CGI/1.1", track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["SERVER_PROTOCOL\x00"], request.Proto, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["SERVER_SOFTWARE\x00"], "FrankenPHP", track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["HTTP_HOST\x00"], request.Host, track_vars_array, thread) // added here, since not always part of headers
+
+	// These values are always empty but must be defined:
+	registerTrustedVar(knownServerKeys["AUTH_TYPE\x00"], "", track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["REMOTE_IDENT\x00"], "", track_vars_array, thread)
+
+	// These values are already present in the SG(request_info), so we'll register them from there
+	C.frankenphp_register_variables_from_request_info(
```

Comment:
While creating some fuzzing tests I noticed that all values in the 'request_info' are currently also registered [directly from go](https://github.com/dunglas/frankenphp/blob/ed3703a16a58355a11b17c4d55fd6552a964954d/frankenphp.go#L410) and their length is calculated with [strlen](https://github.com/dunglas/frankenphp/blob/ed3703a16a58355a11b17c4d55fd6552a964954d/frankenphp.c#L677) (in main). They could therefore potentially be truncated if they contain `\x00`. I'm not sure if this even is an issue or if a string coming from net/http can even contain `\x00`., but it will make the fuzzing tests fail.
@dunglas Are the values in `SG(request_info)` necessary for some internal PHP logic? Should we sanitize the null termination before doing `C.Cstring()` or does it not matter?

Reply:
More spefiically, go allows null bytes in `request.URL.Path` and will convert `%00` to `\x00`. I added some validation and [fuzzing](https://go.dev/doc/security/fuzz/) tests for request inputs. It's probably better to outright reject these paths

Reply:
Indeed let's reject them and send a 400.